devops/providers/azure-devops/templates/infrastructure/README.md

```diff
@@ -0,0 +1,204 @@
+# Infrastructure Continuous Integration and Deployment Pipeline
+
+Azure Pipeline build and release templates responsible for operationalizing infrastructure resources with a focus on repeatability and test automation.
+
+## Introduction
+
+Azure Devops is the common fabric between developers, process and products to enable continuous delivery of value to customers.
+
+Cobalt's devops release flow follows a git commit driven model. Meaning that azure pipelines automatically builds, validates and release Cobalt templates every pull request and commit to your source git repository, essentially applying the GitOps pattern to the Azure PaaS services.
+
+This document outlines how Cobalt's CI/CD Azure Pipeline templates will be used to meet the devops related use-cases for deployment operators working on Cobalt.
+
+## Deployment Topology
+
+This graphic shows the CI/CD workflow topology needed by our enterprise customers to deploy infrastructure to Azure. The journey starts with a git commit to the IAC repo. This workflow validates and deploys the desired infrastructure state across the azure environments.
+
+![infrastructure_reference_cicd_flow](https://user-images.githubusercontent.com/7635865/60440330-3a039580-9bda-11e9-8af6-a883739c2d87.jpg)
+
+## In Scope
+
+- Azure DevOps build template
+- Sequence diagram showing the journey with sourcing, testing, building and releasing IAC(Infra-as-code) templates across azure environment stages.
+
+## Out of Scope
+
+- Observability
+- Monitoring
+- Azure Boards
+- Automating the provisioning of the azure devops pipeline prerequisites
+- Automated load testing
+- Chaos testing
+- Active Directory Setup
+- Multi-region deployments
+- Build Agent Pool Setup
+- Remote terraform modules for auto scaling and monitoring rules
+
+## Customers
+
+- **Technical Operators**: This template uses a service principal to carry out the terraform resource provisioning. The deployment service principal will need contributor role level permissions on the cloud resources.
+
+    Provisioning templates that modify role assignments will require owner level access to the target resource(s).
+
+## Security
```

Comment:
Nit: I think the document may flow a bit better if you move this to a later section of the document so that it generally goes high level > low level as you read further. I gave a proposal below.